### DISCUSSION TITLE
Is there a way to embed the wasm inside the js file?

### DISCUSSION THREAD

**Discussion by BlackAsLight** [wasm-bindgen/wasm-bindgen#3601] [Q&A] [answered]
### Summary

Essentally the title. I'm wondering if there is a way I can have the wasm be embeded in the javascript file so its one thing and doesn't need to perform a fetch to a wasm file.

### Additional Details

I'm not looking for the wasm to be javascript. The environment I'm working in does support wasm, but I'd like to have it all bundled together as one file.

**Comment by daxpedda** [comment]
[`WebAssembly.instantiate()`](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/instantiate) takes an `ArrayBuffer`, so you could encode your Wasm file into let's say base64 and then load it into an `ArrayBuffer` and hand it off to `WebAssembly.instantiate()`.

**Reply by BlackAsLight** [reply]
The wasm code I have requires the second argument, `import`, for it to work, but I don't know how I'd abstract this from the app.js file to combine it with the app_bg.wasm file in base64 form.

**Reply by BlackAsLight** [reply]
nvm. I see that I can do `initSync(new WebAssembly.Module(arrayBuffer))`

**Comment by BlackAsLight** [comment, marked as answer]
I ended up making a working example that lets you bundle it all together as one big file. https://github.com/BlackAsLight/BundleWasm
